### PR TITLE
xplat: add package creator

### PR DESCRIPTION
**Pull request by obastemur**

### CODE REVIEW

**Comment by obastemur** [comment]
Adds a script file that automates xplat package creation.

**Comment by obastemur** [comment]
@dilijev don't we have Tab check exception for Makefile ?

**Comment by obastemur** [comment]
+ https://ci2.dot.net/job/Microsoft_ChakraCore/job/release_2.0/job/ubuntu_check_eol_prtest/118/console doesn't make sense

**Comment by obastemur** [comment]
Looks like https://github.com/Microsoft/ChakraCore/commit/c3c3db5086271d590413062a2dd843927213e782 brakes CR on some of the files.  

I had rebased my branch to latest before adding the later commits

**Comment by obastemur** [comment]
@leirocks FYI https://github.com/Microsoft/ChakraCore/commit/c3c3db5086271d590413062a2dd843927213e782#diff-24781bfa8dd1a4dfe06fdd122676b9c1

File: bin/ch/ch.cpp [hunk 1]
```diff
@@ -94,7 +94,12 @@ void __stdcall PrintUsage()
 
 void __stdcall PrintChVersion()
 {
-    wprintf(_u("%s version %d.%d.%d.0\n"), hostName, CHAKRA_CORE_MAJOR_VERSION, CHAKRA_CORE_MINOR_VERSION, CHAKRA_CORE_PATCH_VERSION);
+#if CHAKRA_CORE_VERSION_RELEASE
+    wprintf(_u("%s version %d.%d.%d.0\n"),
+#else
+    wprintf(_u("%s version %d.%d.%d.0-beta\n"),
```
Comment (jianchun) [inline bin/ch/ch.cpp]: >-beta [](start = 37, length = 5)

Have you synced this "-beta" suffix with others (consistency with messages / terms elsewhere, if we publish prereleases?)

File: build.sh [hunk 2]
```diff
@@ -142,48 +151,9 @@ while [[ $# -gt 0 ]]; do
...
-                    --disable-icuio --disable-layout\
-                    CXXFLAGS="-fPIC" CFLAGS="-fPIC"
-
-            ERROR_EXIT "rm -rf ${ABS_DIR}/icu/source/output/"
-            make STATICCFLAGS="-fPIC" STATICCXXFLAGS="-fPIC" STATICCPPFLAGS="-DPIC" install
-            ERROR_EXIT "rm -rf ${ABS_DIR}/icu/source/output/"
-            cd "${ABS_DIR}/../"
-        fi
-        ICU_PATH="-DCC_EMBED_ICU_SH=1"
+        SHOULD_EMBED_ICU=1
```
Comment (jianchun) [inline build.sh]: >SHOULD_EMBED_ICU [](start = 8, length = 16)

Should this (SHOULD_EMBED_ICU) have a default value like others?

File: build.sh [hunk 3]
```diff
@@ -325,6 +295,10 @@ while [[ $# -gt 0 ]]; do
         WB_ARGS=${WB_ARGS// /;}  # replace space with ; to generate a cmake list
         ;;
 
+    -y | -Y)
+        ALWAYS_YES=1
```
Comment (jianchun) [inline build.sh]: >ALWAYS_YES [](start = 8, length = 10)

Should this (ALWAYS_YES) have a default value like others?

**Comment by jianchun** [comment]
:shipit:

File: bin/ch/ch.cpp (hunk 1 above)
Comment (obastemur) [inline bin/ch/ch.cpp]: /cc @liminzhu @dilijev ?

Comment (dilijev) [inline bin/ch/ch.cpp]: I think this is fine. @liminzhu should handle checking over the various documentation, but I think at the moment it's all in discussion and maintained by @obastemur unless I missed something being published.

File: bin/ch/ch.cpp [hunk 4]
```diff
@@ -94,7 +94,12 @@ void __stdcall PrintUsage()
 
 void __stdcall PrintChVersion()
 {
-    wprintf(_u("%s version %d.%d.%d.0\n"), hostName, CHAKRA_CORE_MAJOR_VERSION, CHAKRA_CORE_MINOR_VERSION, CHAKRA_CORE_PATCH_VERSION);
+#if CHAKRA_CORE_VERSION_RELEASE
```
Comment (dilijev) [inline bin/ch/ch.cpp]: See comments in `ChakraCoreVersion.h` but "RELEASE" configuration is `CHAKRA_CORE_VERSION_RELEASE && !CHAKRA_CORE_VERSION_PRERELEASE`. If `CHAKRA_CORE_VERSION_RELEASE && CHAKRA_CORE_VERSION_PRERELEASE`, then we're in prerelease mode. If both are 0, then we're in dev mode. Both of the latter could be considered `beta` for publication purposes, unless you want to make a distinction between `-dev` and `-preview` (or `-rc` and `-beta`). But IMO `-beta` is fine.

File: tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt [hunk 5]
```diff
@@ -0,0 +1,295 @@
+THIRD-PARTY SOFTWARE NOTICES AND INFORMATION
+Do Not Translate or Localize
+
+This ChakraCore package is based on or incorporates material from the projects listed below (Third Party IP). The original copyright notice and the license under which Microsoft received such Third Party IP, are set forth below. Such licenses and notices are provided for informational purposes only. Microsoft licenses the Third Party IP to you under the licensing terms for the ChakraCore NuGet package. Microsoft reserves all other rights not expressly granted under this agreement, whether by implication, estoppel or otherwise.
```
Comment (dilijev) [inline tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt]: Wrap to 80 columns for easier reading in text editors?

File: tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+=========================================
+END OF ChakraCore  NOTICES, INFORMATION, AND LICENSE
+
+%% Intel VTune Amplifier XE NOTICES, INFORMATION, AND LICENSE BEGIN HERE
+=========================================
```
Comment (dilijev) [inline tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt]:
```
+=========================================
 +END OF ChakraCore  NOTICES, INFORMATION, AND LICENSE
 +
 +%% Intel VTune Amplifier XE NOTICES, INFORMATION, AND LICENSE BEGIN HERE
 +=========================================
```

Is this a standard format for delimiting such notices?

File: tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+=========================================
+END OF Intel VTune Amplifier XE NOTICES, INFORMATION, AND LICENSE
+
+%% File copyright Microsoft and Geoff Norton NOTICES, INFORMATION, AND LICENSE BEGIN HERE
```
Comment (dilijev) [inline tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt]: Which file?

File: tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+=========================================
+Copyright (c) Microsoft
+Copyright (c) Geoff Norton
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the ""Software""), to deal
```
Comment (dilijev) [inline tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt]: > `(the ""Software"")`

Why double quotes? Various places, here and elsewhere.

File: tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+Copyright†(C)†2009†Free†Software†Foundation,†Inc.
+
+Everyone†is†permitted†to†copy†and†distribute†verbatim†copies†of†this
+license†document,†but†changing†it†is†not†allowed.
+
+This†GCC†Runtime†Library†Exception†("Exception")†is†an†additional
+permission†under†section†7†of†the†GNU†General†Public†License,†version
+3†("GPLv3").†It†applies†to†a†given†file†(the†"Runtime†Library")†that
+bears†a†notice†placed†by†the†copyright†holder†of†the†file†stating†that
+the†file†is†governed†by†GPLv3†along†with†this†Exception.
```
Comment (dilijev) [inline tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt]: Corrupted spaces? (`†` is `\u2020`)

File: tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
+TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+PERFORMANCE OF THE DATA FILES OR SOFTWARE.
+
+Except as contained in this notice, the name of a copyright holder
+shall not be used in advertising or otherwise to promote the sale,
+use or other dealings in these Data Files or Software without prior
+written authorization of the copyright holder.
+=========================================
+END OF icu4c NOTICES, INFORMATION, AND LICENSE
```
Comment (dilijev) [inline tools/XPlatInstall/BINARY-DIST-ONLY-NOTICES.txt]: Is all of this file approved by legal? All licenses are compatible with our MIT license for binary distribution?

**Comment by obastemur** [comment]
@dilijev the third party binary license file is from code scan / legal team and I store it as is. This file is not for the repository for particularly for the xplat binary.

File: build.sh (hunk 2 above)
Comment (obastemur) [inline build.sh]: Added

File: build.sh (hunk 3 above)
Comment (obastemur) [inline build.sh]: Added

**Comment by obastemur** [comment]
This PR includes commits from #2880

**Comment by obastemur** [comment]
Also updated license file (look).

@jianchun @dilijev Thanks for the review

**Comment by dilijev** [comment]
> Also updated license file (look).

Looks like the corrupted spaces are fixed. I assume the rest (double quotes and so on) are editorial and this is just the way the file was received/approved?

**Comment by obastemur** [comment]
Yes, that's how it is received

**Pull request merged**